### ISSUE TITLE
Why are the signatures OCI Images rather than OCI Artifacts?

### ISSUE THREAD

**Issue by TBBle** [sigstore/cosign#913]
**Question**

Why are the signatures OCI Images rather than OCI Artifacts?

I've been poking through the docs and website for a while, but couldn't track down an answer, or even a discussion.

It seems to me that the current [OCI Images created by this tool](https://github.com/sigstore/cosign/blob/2594f7acc849644a6bd05e5f3fa567ac87056b85/specs/SIGNATURE_SPEC.md?plain=1#L183-L201) ([relevant source](https://github.com/sigstore/cosign/blob/2594f7acc849644a6bd05e5f3fa567ac87056b85/pkg/oci/static/options.go#L39-L41)) are not valid per [the OCI Image spec](https://github.com/opencontainers/image-spec/blob/main/config.md) because the layers are `application/vnd.dev.cosign.simplesigning.v1+json`, which is not covered by the [layer spec](https://github.com/opencontainers/image-spec/blob/main/layer.md). This means that, for example, if you pointed a container runtime at the uploaded image, its initial checks would pass (i.e. config `application/vnd.oci.image.config.v1+json` is recognised) but then the layer itself is in an unknown format. (If there's no actual OCI Image Config object being uploaded, then it'd probably barf on looking for that too... It's `<omitted for brevity>` in the spec, but glancing at the code, I don't think a config object is linked at all? I did see a bug report with a SHA reference for a config object though, but I don't know what that would be...)

And it's pretty explicitly stated in [the OCI Artifact guidelines](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#defining-a-unique-artifact-type):
> Note: The `config.mediaType` of `application/vnd.oci.image.config.v1+json` is reserved for artifacts intended to be run and instanced by docker, containerd and other OCI Image runtimes and tool chains.

I was clued in to this oddity by [this change in quay](https://github.com/quay/quay/pull/877/files) where you can see it specifying the layer types allowed for Helm OCI Artifacts, and bizarrely specifying that _OCI Images_ have a layer type of `application/vnd.dev.cosign.simplesigning.v1+json` (which I _assume_ is additive in their code-base)

This projects seems like a prime candidate for being an [OCI Artifact](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#defining-oci-artifact-types), and defining your own config mediaType, e.g., `application/vnd.dev.cosign.config.v1` (assuming there's actual config data to be stored), would meet the desired criteria that anything pulling such a manifest knows immediately if it can handle it or not. It would also allow projects like quay which require _explicit_ OCI Artifact opt-in, to support opt-in distinctly for cosign as its own artifact type, rather than a weird appendage of OCI Images.

**Comment by TBBle** [comment]
Still a thing, but pending on https://github.com/docker/roadmap/issues/135 before any changes can be made on this.

**Comment by imjasonh** [comment]
In the last ~month, the OCI working group for reference types has proposed and accepted https://github.com/opencontainers/distribution-spec/pull/335 and https://github.com/opencontainers/image-spec/pull/934 which describe an OCI-official method both for describing and querying artifacts-that-reference-things (e.g., signatures), _and_ a fallback naming mechanism for registries that don't yet support the new official method.

Over time I expect cosign to adopt this new method, hopefully/probably via support in go-containerregistry, so the ecosystem of things that create signatures/etc (e.g., Tekton Chains) can do so in a common, OCI-specified way, and rely on the fallback naming mechanism to push to and pull from older registries.

**Comment by TBBle** [comment]
DockerHub has now [gained support for OCI Artifacts](https://www.docker.com/blog/announcing-docker-hub-oci-artifacts-support/), so that's one fewer roadblocks for the changes @imjasonh described above.

**Comment by dlorenc** [comment]
Yes! We should make the migration.

**Comment by znewman01** [comment]
Duplicate of https://github.com/sigstore/cosign/issues/1397 which has more activity recently
